pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -1820,7 +1820,6 @@ func TestInsufficientCapacityNodeDaemonLaunchesCriticalPod(t *testing.T) {
 		ds := newDaemonSet("critical")
 		ds.Spec.UpdateStrategy = *strategy
 		ds.Spec.Template.Spec = podSpec
-		setDaemonSetCritical(ds)
```

Comment:
If I understand correctly, this tests is about creating a critical daemon set. So, why not keeping this line to create a critical daemon set? In that case, I think the second for loop that you have added in this PR can be removed.

Reply:
The first loop is for checking the daemon set pod was not scheduled and the second one checks critical daemonset. I could remove this if it is not necessary to check the result when daemonset is not critical.

Reply:
Yeah, I thought it was not necessary to check that non-critical daemonsets are not scheduled, but let's leave the check in.